### DISCUSSION TITLE
Can an lsp client intercept the default behaviour of notifications from the server

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#579] [Extension Development QnA] [answered]
The lsp server for Clojure sends `textDocument/publishDiagnostics` notifications for various errors. It makes a message pop up to the user. Can the lsp client change this behaviour? I tried this in Calva's lsp client:

```typescript
  client.onNotification('textDocument/publishDiagnostics', (params) => {
    console.log('Notification', params);
  });
```

The error message shows before the logging happens. This diagnostic is not handled by the `handleDiagnostics` middleware function either...

**Comment by isidorn** [comment, marked as answer]
I do not think this is possible.

**Reply by PEZ** [reply]
Thanks!
